app/standalone_test.go

```diff
@@ -196,6 +198,83 @@ func Test_doManualUpdate_existingFile_updateSuccess(t *testing.T) {
 	assert.NoError(t, err)
 }
 
+func TestDoManualUpdateArtifactV3Dependencies(t *testing.T) {
+	// setup
+	deviceType := zeroLengthDeviceTypeFile(t)
+	defer os.Remove(deviceType)
+
+	artifactProvides := &tests.ArtifactProvides{
+		ArtifactName:  "testName",
+		ArtifactGroup: "testGroup",
+	}
+	artifactDepends := &tests.ArtifactDepends{
+		ArtifactName:      []string{"OldArtifact"},
+		ArtifactGroup:     []string{"testGroup"},
+		CompatibleDevices: []string{"qemux86-64"},
+	}
+	typeInfoDepends := &map[string]interface{}{
+		"testKey": "testValue",
+	}
+
+	artifactStream, err := tests.CreateTestArtifactV3("test", "gzip",
+		artifactProvides, artifactDepends, nil, typeInfoDepends)
+	require.NoError(t, err)
+	require.NotNil(t, artifactStream)
+
+	tmpdir, err := ioutil.TempDir("", "mendertest")
+	require.NoError(t, err)
+	defer os.RemoveAll(tmpdir)
+
+	f, err := ioutil.TempFile("", "update")
+	require.NoError(t, err)
+	defer os.Remove(f.Name())
+
+	_, err = io.Copy(f, artifactStream)
+	assert.NoError(t, err)
+	f.Close()
+
+	// test
```

Comment:
Suggestion: Not necessary to follow this up. But one can provide a setup function to the test, along with a function for cleanup, so it turns into something like:
```
testData, cleaup := testSetup()
defer cleanup()
```